packages/cli/src/ui/hooks/useCompletion.ts

```diff
@@ -300,21 +302,32 @@ export function useCompletion(
         respectGitIgnore?: boolean;
         respectGeminiIgnore?: boolean;
       },
+      signal: AbortSignal,
       currentRelativePath = '',
       depth = 0,
-      maxDepth = 10, // Limit recursion depth
-      maxResults = 50, // Limit number of results
+      maxDepth = 8, // Reduced from 10 for better performance
+      maxResults = 30, // Reduced from 50 for better performance
     ): Promise<Suggestion[]> => {
+      // Check if operation was aborted
+      if (signal.aborted) {
+        return [];
+      }
+
       if (depth > maxDepth) {
         return [];
       }
 
       const lowerSearchPrefix = searchPrefix.toLowerCase();
       let foundSuggestions: Suggestion[] = [];
       try {
+        // Check abort signal before reading directory
+        if (signal.aborted) {
+          return [];
+        }
         const entries = await fs.readdir(startDir, { withFileTypes: true });
         for (const entry of entries) {
-          if (foundSuggestions.length >= maxResults) break;
+          // Check abort signal frequently during intensive operations
```

Comment:
nit: move  this check outside the for loop as it only needs to occur after the async operation.